pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -965,14 +988,30 @@ func (service *SecurityPolicyService) buildPeerTags(obj *v1alpha1.SecurityPolicy
 		peerTags = append(peerTags, tag)
 	}
 
-	// In non-VPC network, there is no need to add is_project_shared tag for rule peer groups
-	if isVpcEnabled(service) {
-		peerTags = append(peerTags,
-			model.Tag{
-				Scope: String(common.TagScopeProjectGroupShared),
-				Tag:   groupSharedTag,
-			},
-		)
+	// In non-VPC network, there is no need to add is_project_shared or is_infra_shared tag for rule peer groups
+	if isVPCEnabled(service) {
+		if infraGroupShared {
+			peerTags = append(peerTags,
+				model.Tag{
+					Scope: String(common.TagScopeNSXShareCreatedFor),
+					Tag:   String(common.TagValueShareCreatedForInfra),
+				},
+			)
+		} else if projectGroupShared {
+			peerTags = append(peerTags,
+				model.Tag{
+					Scope: String(common.TagScopeNSXShareCreatedFor),
+					Tag:   String(common.TagValueShareCreatedForProject),
+				},
+			)
+		} else {
+			peerTags = append(peerTags,
+				model.Tag{
+					Scope: String(common.TagScopeNSXShareCreatedFor),
+					Tag:   String(common.TagValueShareNotCreated),
+				},
+			)
+		}
```

Comment:
I tried to merge these two tag values into one, but it has some problem.
The problem will come up from here:
https://github.com/vmware-tanzu/nsx-operator/pull/670/files#diff-53bc6d3b74f0d34b4187204c1bcf7ce4afe93c7f73e60b5de43c40dc44b88af0R721

Where it tries to initialize the infra store, Since the scope for both infra group share and project group share is the same:  common.TagScopeNSXShareCreatedFor, if the tag value is the same as well. It can not determine these groups/shares are created from infra level or project level.  That is why we need tag value to tell us more differences.